src/FileFormat/ZipArchiveHandler.vala

```diff
@@ -259,6 +259,7 @@ public class Akira.FileFormat.ZipArchiveHandler : GLib.Object {
         Archive.Result last_result;
 
         while ((last_result = archive.next_header (out entry)) == Archive.Result.OK) {
+            entry.set_perm (0644);
```

Comment:
In libarchive basic examples it does this too:

https://github.com/libarchive/libarchive/wiki/Examples#A_Basic_Write_Example

> archive_entry_set_perm(entry, 0644);

so probably it's ok to do that. Not sure

Reply:
When unzipping, the permissions for the unzipped files (0000) are changed with the new permissions (0644) so ​​that they can be accessed.

Reply:
I put it there for the people who saved the file and compiled Akira like me so they can then open when the file doesn't have permissions, you don't read the comments? (I can't blame you, GitHub doesn't update them, you have to refresh the page).